plutus-core/plutus-ir/src/PlutusIR/Transform/Inline/Inline.hs

```diff
@@ -248,7 +248,7 @@ processSingleBinding body = \case
                 void $ modify' $
                     extendVarInfo
                         n
-                        (MkVarInfo s (Done (dupable processedRhs)) varArity bodyToCheck)
+                        (MkVarInfo s processedRhs arity bodyToCheck)
```

Comment:
see below, should be an `InlineTerm`, not just `Term`.

Reply:
OK looking at  [Dupable](https://github.com/input-output-hk/plutus/blob/2d0a953ffcdce0975595c3c9422ff1202d7ae63e/plutus-core/plutus-core/src/PlutusCore/Rename/Internal.hs#L46) again I've apparently used it wrong. And if we are using `Dupable` we don't need to rename, because that's the point. So it seems we should put it in as an `InlineTerm` and call `liftDupable` when we do the substitution. I think we may not need to rename in the unconditional inliner too in this case.

Reply:
The point of `Dupable` is to signal that "this is a term that you need to rename when you use it". We indeed shouldn't be matching on the `Dupable` constructor, that does rather defeat the point! Sorry I missed that before!

Reply:
I don't think using `Dupable` on `varBody` makes sense. E.g., for `f = \ x y -> x + y`, renaming `varBody` means renaming `x + y`, which is not a great idea because now `x` and `y` are free. You need to do the renaming of `f` i.e., `varDef`.

Reply:
I have since found that the renamer doesn't touch free variables. It's more of a problem that `varBody`  can't be renamed, because it looks like a bunch of free variables, and thus won't be touched. See [this comment](https://github.com/input-output-hk/plutus/pull/5395#discussion_r1238789249).

Reply:
What's wrong with the way `varBody` is renamed? The renamer won't touch the free variables, which is correct, but it will rename any bound variables in there, which is what we want?